scipy/linalg/_decomp_cholesky.py

```diff
@@ -374,3 +376,109 @@ def cho_solve_banded(cb_and_lower, b, overwrite_b=False, check_finite=True):
+        factorization.
+    overwrite_R: bool, default: False
+        If set to True, the provided Cholesky factor array will be modified in-place.
+        in-place instead of creating a new array.
+    overwrite_z : bool, default: False
+        If set to True, the provided update array will be modified will be
+        modified in-place.
+    eps : non-negative float, optional
+        A tolerance for a check that the updated matrix is positive definite.
+        With exact arithmetic this value would be zero, but with finite arithmetic,
+        the value must be positive for a conservative check.
+        The default is `np.finfo(dtype).eps`, where `dtype` is the result type
+        of `R` and `Z`.
+    check_finite : bool, default: False
+        Whether to check that the input matrix contains only finite numbers.
+        Disabling may give a performance gain, but may result in problems
+        (crashes, non-termination) if the inputs do contain infinities or NaNs.
+        Default is True.
+
+    Returns
+    -------
+    D : (N, N) ndarray
+        The resulting modified Cholesky factor.
+
+    References
+    ----------
+    .. [1] : C.T. Pan, "A modification to the LINPACK downdating algorithm",
+       BIT Numerical Mathematics, Vol.30(4), 1990, :doi:`10.1007/BF01933218`
+
+    """
+    # Input validation and standardization
+    R1 = _asarray_validated(R, check_finite=check_finite)
+    z1 = _asarray_validated(z, check_finite=check_finite)
+
+    if R1.dtype != z1.dtype:
+        dtype = np.result_type(R1.dtype, z1.dtype)
+        R1 = R1.astype(dtype)
+        z1 = z1.astype(dtype)
+
+    R = R1 if overwrite_R or _datacopied(R1, R) else R1.copy()
+    z = z1 if overwrite_z or _datacopied(z1, z) else z1.copy()
+
+    if R.ndim != 2:
+        raise ValueError("The Cholesky factor `R` must have two dimensions.")
+    if z.ndim != 1:
+        raise ValueError("The update vector `z` must have one dimension.")
+
+    m, n = R.shape
+    if m != n:
+        raise ValueError('The Cholesky factor `R` must be square.')
+    if m != z1.size:
+        raise ValueError('The size of update vector `z` must equal the number '
+                         'of rows of Cholesky factor `R`.')
+
+    if np.iscomplexobj(R):
+        raise ValueError("Complex input is not supported.")
+
+    if eps is None:
+        eps = n * np.finfo(R1.dtype).eps
+    if eps < 0:
+        raise ValueError("`eps` must be non-negative.")
+
+    R = R.T if lower else R
+
+    # Initialization
+    alpha, beta = empty_like(z), empty_like(z)
+    alpha[-1], beta[-1] = 1., 1.
+    sign = -1 if downdate else 1
+
+    # Main algorithm
+    for r in range(n):
+        a = z[r] / R[r, r]
+        alpha[r] = alpha[r - 1] + sign * a**2
+        if alpha[r] < eps:  # numerically zero or negative
+            raise LinAlgError('The Cholesky factor becomes nonpositive'
+                              f'with this downdate at step {r}')
```

Comment:
Is it important to mention the step number here? What does the user do with this information (unless they inspect the code to determine what the steps are)? Shall I change this to: "The updated/downdated Cholesky factor is not positive definite."?